### PR TITLE
V2 Updates

### PR DESCRIPTION
**Pull request by konradpabjan**
## Overview

- Updates to the `README` to document compat differences between `v1` and `v2`
   - 🖌 Rendered [README](https://github.com/actions/download-artifact/tree/konradpabjan/updates-to-v2) 🎨
- Update all NPM packages to the latest versions (gets rid of some security alerts)
- Run `npm run release` to update `dist/index.js` so the latest packages get used everywhere

## Testing

We have E2E tests for each PR 😀 and it's looking good
Extra sanity check with some larger uploads & downloads: https://github.com/konradpabjan/artifact-test/runs/634397827?check_suite_focus=true

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,6 +44,22 @@ steps:
   run: ls -R
   working-directory: path/to/artifact
 ```
+
+## Compatibility between `v1` and `v2`
+
+When using `download-artifact@v1`, a directory denoted by the name of the artifact would be created if the `path` input was not provided. All of the contents would be downloaded to this directory.
```
Comment (yacaovsnc) [inline README.md:50]: Does the existence of `path` input matter?  Is a folder with artifact name always created in `v1`?

File: README.md [hunk 2]
```diff
@@ -44,6 +44,22 @@ steps:
...
+```
+   current/working/directory/
+      my-artifact/
+          ... contents of my-artifact
+```
+
+With `v2`, there is no longer an extra directory that is created if the `path` input is not provided. All the contents are downloaded to the current working directory.
```
Comment (yacaovsnc) [inline README.md:57]: This really isn't about the existence of `path` input though.  In v2, we just do not create the artifact root folder.  You could say something like "to emulate the v1 behavior, use `path` parameter and specific the where you want to download the artifact contents."

An example with `path` input would help too.

File: README.md (hunk 1 above)
Comment (konradpabjan) [reply README.md:50]: The `path` does matter, the folder with the name is only created if the `path` is not provided for `v1`. 

`v1` with `path` provided: https://github.com/konradpabjan/artifact-test/runs/634397827?check_suite_focus=true#step:7:15
`v1` with `path` not provided: https://github.com/konradpabjan/artifact-test/runs/634463778?check_suite_focus=true#step:7:17

File: README.md (hunk 2 above)
Comment (konradpabjan) [reply README.md:57]: I added an example of how users can achieve the same behavior as `v1` if they set the `path` input to the name of the artifact.

**Pull request merged**